crates/turborepo-lib/src/turbo_json/loader.rs

```diff
@@ -147,9 +170,19 @@ impl TurboJsonLoader {
                     load_from_root_package_json(&self.repo_root, root_turbo_json, package_json)
                 }
             }
-            Strategy::Workspace { packages } => {
+            Strategy::Workspace {
+                packages,
+                micro_frontends_configs,
+            } => {
                 let path = packages.get(package).ok_or_else(|| Error::NoTurboJSON)?;
-                load_from_file(&self.repo_root, path)
+                let should_inject_proxy_task = micro_frontends_configs
+                    .as_ref()
+                    .map_or(false, |configs| configs.contains_package(package.as_str()));
+                if should_inject_proxy_task {
+                    load_from_file_with_proxy(&self.repo_root, path)
```

Comment:
Could we make this a method on `TurboJson`, i.e. `with_proxy()` that injects the proxy task?